controllers/service_controller_test.go

```diff
@@ -0,0 +1,222 @@
+package controllers
+
+import (
+	"context"
+	"github.com/3scale-labs/authorino/api/v1beta1"
+	"github.com/3scale-labs/authorino/pkg/cache"
+	"gotest.tools/assert"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"testing"
+)
+
+var (
+	service = v1beta1.Service{
```

Comment:
Maybe we could introduce in the future something like Factories to create these kind of objects